File: internal/entity/chart.go
package entity

import (
    _ "gorm.io/gorm"
    "time"
)

type Chart struct {
    ID         int64 `gorm:"primaryKey"`
    IDOffice   int64
    WeeksDay   string
    ChartsTime *time.Time `gorm:"type:time"`
    Chart      int
    Office     Office `gorm:"foreignKey:IDOffice"`
}

func (co *Chart) TableName() string {
    return "charts_offices"
}

File: src/main/java/com/vtb/hack/hack/entity/Chart.java
package com.vtb.hack.hack.entity;

import java.sql.Time;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "charts_offices")
@Data
public class Chart extends BaseEntity {
    @Column(nullable = false)
    private String weeksDay;

    private Time chartsTime;

    private int chart;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "office_id", nullable = false)
    private Office office;
}
